src/core/autobridgecalculator.js

```diff
@@ -42,18 +42,42 @@ const NULL_AMOUNT = Utils.normalizeAmount('0');
  * @return {Array}
  */
 
-AutobridgeCalculator.prototype.calculate = function() {
-  const oldMode = Amount.strict_mode;
-  Amount.strict_mode = false;
+AutobridgeCalculator.prototype.calculate = function(callback) {
 
-  let legOnePointer = 0;
-  let legTwoPointer = 0;
+  const legOnePointer = 0;
+  const legTwoPointer = 0;
 
   const offersAutobridged = [];
 
   this.clearOwnerFundsLeftover();
 
+  this._calculateInternal(legOnePointer, legTwoPointer, offersAutobridged,
+    callback);
+};
+
+AutobridgeCalculator.prototype._calculateInternal = function(
+  legOnePointer_, legTwoPointer_, offersAutobridged, callback
+) {
+  this._oldMode = Amount.strict_mode;
+  Amount.strict_mode = false;
```

Comment:
I think in any case we must have way to switch off checks for computationally expensive cases like this.